### PR TITLE
Rocket details view

### PR DESCRIPTION
**Pull request by j7uu**

### CODE REVIEW

**Comment by j7uu** [comment]
It's not done yet, but there was too much of it just sitting on my local

File: client/src/views/rocket-details-view.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                            {rocketData?.Name}
+                        </Typography>
+                    </Card>
+				</Grid>
+                    <Box sx={{ width: '100%' }}>
+                        <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
+                            <Tabs value={value} onChange={handleChange} aria-label="basic tabs example">
+                                <Tab label="Item One" />
+                                <Tab label="Item Two"  />
+                                <Tab label="Item Three" />
```
Comment (klemie) [inline client/src/views/rocket-details-view.tsx:116]:
```suggestion
                                <Tab label="Details" />
                                <Tab label="Components"  />
                                <Tab label="Missions" />

```

File: client/src/views/rocket-details-view.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                <Grid item>
+					<Card>
+                        <Typography marginX={"2rem"} marginY={"1rem"} align='left' variant='h6'>
+                            {rocketData?.Name}
+                        </Typography>
+                    </Card>
+				</Grid>
+                    <Box sx={{ width: '100%' }}>
+                        <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
+                            <Tabs value={value} onChange={handleChange} aria-label="basic tabs example">
```
Comment (klemie) [inline client/src/views/rocket-details-view.tsx:113]:
```suggestion
                            <Tabs value={value} onChange={handleChange}>

```

File: client/src/views/rocket-details-view.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                        </Box>
+                        <TabPanel value={value} index={0}>
+                            Item One
+                        </TabPanel>
+                        <TabPanel value={value} index={1}>
+                            Item Two
+                        </TabPanel>
+                        <TabPanel value={value} index={2}>
+                            Item Three
+                        </TabPanel>
```
Comment (klemie) [inline client/src/views/rocket-details-view.tsx:127]:
```suggestion
                        <TabPanel value={value} index={0}>
                             Details
                        </TabPanel>
                        <TabPanel value={value} index={1}>
                            Components
                        </TabPanel>
                        <TabPanel value={value} index={2}>
                            Missions
                        </TabPanel>

```

**Review by klemie — CHANGES_REQUESTED** [review]
W

File: client/src/views/rocket-details-view.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            {...other}
+        >
+        {value === index && (
+            <Box sx={{ p: 3 }}>
+                <Typography>{children}</Typography>
+            </Box>
+        )}
+        </div>
+        );
+    }
```
[the remark this answers is not in the document]
Comment (j7uu) [reply client/src/views/rocket-details-view.tsx:37]: twas necessary

File: client/src/utils/view-provider.tsx [hunk 5]
```diff
@@ -70,24 +76,28 @@ export default function ViewProvider(props: ViewProviderProps) {
...
     useEffect(() => {
         console.log(currentViewKey);
     }, [currentViewKey])
     return (
-        <div className='app'>
-            {currentViewKey !== (TELEMETRY_KEY || FLIGHT_KEY || RECOVERY_KEY || FLIGHT_REPORT_KEY)  && currentView(ROCKET_SELECT_KEY)}
+		<div className='app'>
+
+            {currentViewKey === (ROCKET_SELECT_KEY)  && currentView(ROCKET_SELECT_KEY)}
+			{currentViewKey === (ROCKET_DETAILS_KEY) && currentView(ROCKET_DETAILS_KEY)}
```
[the remark this answers is not in the document]
Comment (j7uu) [reply client/src/utils/view-provider.tsx:100]: W

**Review by klemie — APPROVED** [review]

**Pull request merged**